### PR TITLE
feat(functions): add support for deno on arm64 Linux

### PR DESCRIPTION
**Pull request by AaronDewes**
This adds support for linux/arm64 to the Supabase functions CLI by utilizing the unofficial arm64 builds of Deno from https://github.com/LukeChannings/deno-arm64

### CODE REVIEW

**Comment by sweatybridge** [comment]
Hi @AaronDewes , thanks for creating this PR. We are focused on fixing bugs this week since we are in the middle of a launch.

New features like yours will have to wait until next week before we review and merge.

Appreciate your understanding and support.

**Comment by AaronDewes** [comment]
Another issue I noticed with the CLI on arm is that https://github.com/supabase/postgres/pull/179 removed arm64 support for postgres, which means commands that use postgres like `supabase db remote commit` fail with a "container is not running error". I think the error handling should be improved (By adding an unsupported architecture message), or arm64 should be added back.

Are there any plans to re-add arm64 builds for postgres?

As a suggestion on how to re-add this: What I did for some repos I manage was to use a self-hosted arm runner to speed up builds, that may be an option for you too: https://github.com/runcitadel/docker-template/tree/1ccdaa39bd613858784f79e4af0d531a3758390e

At Oracle Cloud, one arm64 VM with 4 cores, 24GB RAM and 200GB storage is available for free: https://www.oracle.com/cloud/free/. So maybe you want to host a self-hosted actions runner there (unless you already have a self-hosted arm64 runner or are already working on another solution, I'm not familiar with your infrastructure).

File: internal/utils/misc.go
Comment (sweatybridge) [inline internal/utils/misc.go:282]:
```suggestion
			// TODO: version pin to official release once available https://github.com/denoland/deno/issues/1846
			assetsUrl = "https://github.com/LukeChannings/deno-arm64/releases/latest/download/"
```

**Review by sweatybridge — APPROVED** [review]
> Are there any plans to re-add arm64 builds for postgres?

Yes, this has been on our backlog for quite some time #41. Let me check with the team and get back with a timeline later this week.

Meanwhile this PR is ready to go. My only suggestion is to add a todo comment.

**Pull request merged**
